File: mac-ui/Blink/Theme/SyntaxTheme.swift
import AppKit

/// TokenType に対応する高コントラストの黒基調シンタックステーマ
enum SyntaxTheme {
    /// UniFFI 生成の TokenType enum から NSColor を返す
    static func color(for tokenType: TokenType) -> NSColor {
        switch tokenType {
        case .keyword:
            // 赤寄りピンク (#FF7B72)
            NSColor(red: 1.000, green: 0.482, blue: 0.447, alpha: 1.0)
        case .string:
            // 明るいグリーン (#7EE787)
            NSColor(red: 0.494, green: 0.906, blue: 0.529, alpha: 1.0)
        case .comment:
            // コメントグレー (#8B949E)
            NSColor(red: 0.545, green: 0.580, blue: 0.620, alpha: 1.0)
        case .type:
            // シアンブルー (#79C0FF)
            NSColor(red: 0.475, green: 0.753, blue: 1.000, alpha: 1.0)
        case .function:
            // パープル (#D2A8FF)
            NSColor(red: 0.824, green: 0.659, blue: 1.000, alpha: 1.0)
        case .number:
            // ゴールド (#F2CC60)
            NSColor(red: 0.949, green: 0.800, blue: 0.376, alpha: 1.0)
        case .operator:
            // ライトグレー (#E6EDF3)
            NSColor(red: 0.902, green: 0.929, blue: 0.953, alpha: 1.0)
        case .punctuation:
            // ミディアムグレー (#C9D1D9)
            NSColor(red: 0.788, green: 0.820, blue: 0.851, alpha: 1.0)
        case .variable:
            // オレンジ (#FFA657)
            NSColor(red: 1.000, green: 0.651, blue: 0.341, alpha: 1.0)
        case .plain:
            // ライトグレー (#E6EDF3)
            defaultTextColor
        }
    }

    /// 背景色 (#0B0D10)
    static var backgroundColor: NSColor {
        NSColor(red: 0.043, green: 0.051, blue: 0.063, alpha: 1.0)
    }

    /// デフォルトテキスト色 (#E6EDF3)
    static var defaultTextColor: NSColor {
        NSColor(red: 0.902, green: 0.929, blue: 0.953, alpha: 1.0)
    }

    /// 行番号色 (#9BA6B2)
    static var lineNumberColor: NSColor {
        NSColor(red: 0.608, green: 0.651, blue: 0.698, alpha: 1.0)
    }
}

File: app/lib/src/theme/syntax_theme.dart
import 'package:flutter/material.dart';

/// GitHub Dark テーマのシンタックスハイライトカラー
class SyntaxTheme {
  SyntaxTheme._();

  static const backgroundColor = Color(0xFF0B0D10);
  static const defaultTextColor = Color(0xFFE6EDF3);
  static const lineNumberColor = Color(0xFF9BA6B2);

  /// FRB生成のDartTokenSpan.tokenType (String) からColorを返す
  /// tokenTypeの値: "keyword", "string", "comment", "type", "function",
  /// "number", "operator", "punctuation", "variable", "plain"
  static Color colorFor(String tokenType) {
    return switch (tokenType) {
      'keyword' => const Color(0xFFFF7B72),
      'string' => const Color(0xFF7EE787),
      'comment' => const Color(0xFF8B949E),
      'type' => const Color(0xFF79C0FF),
      'function' => const Color(0xFFD2A8FF),
      'number' => const Color(0xFFF2CC60),
      'operator' => const Color(0xFFE6EDF3),
      'punctuation' => const Color(0xFFC9D1D9),
      'variable' => const Color(0xFFFFA657),
      _ => defaultTextColor, // "plain" and unknown
    };
  }

  /// コード表示用TextStyle (JetBrains Mono 13pt, lineHeight 1.5)
  static TextStyle get codeStyle => const TextStyle(
        fontFamily: 'JetBrains Mono',
        fontSize: 13,
        color: defaultTextColor,
        height: 1.5,
      );

  /// 行番号表示用TextStyle (JetBrains Mono 11pt)
  static TextStyle get lineNumberStyle => const TextStyle(
        fontFamily: 'JetBrains Mono',
        fontSize: 11,
        color: lineNumberColor,
      );
}
